qiskit/pulse/macros.py

```diff
@@ -77,18 +113,67 @@ def measure(
-                continue
-            if qubit_mem_slots and isinstance(inst, instructions.Acquire):
-                if inst.channel.index in qubit_mem_slots:
-                    mem_slot = channels.MemorySlot(qubit_mem_slots[inst.channel.index])
-                else:
-                    mem_slot = channels.MemorySlot(unused_mem_slots.pop())
-                inst = instructions.Acquire(inst.duration, inst.channel, mem_slot=mem_slot)
-            # Measurement pulses should only be added if its qubit was measured by the user
-            schedule = schedule.insert(time, inst)
+        schedule = combine_schedules(qubits, schedule, default_sched, qubit_mem_slots, measure_group_qubits)
+    return schedule
+
+
+def measure_v2(
+    qubits: List[int],
+    backend=None,
+    target: Optional[Target] = None,
+    meas_map: Optional[Union[List[List[int]], Dict[int, List[int]]]] = None,
+    qubit_mem_slots: Optional[Dict[int, int]] = None,
+    measure_name: str = "measure",
+) -> Schedule:
+    """Return a schedule which measures the requested qubits according to the given
+    instruction mapping and measure map, or by using the defaults provided by the backend.
 
+    By default, the measurement results for each qubit are trivially mapped to the qubit
+    index. This behavior is overridden by qubit_mem_slots. For instance, to measure
+    qubit 0 into MemorySlot(1), qubit_mem_slots can be provided as {0: 1}.
+
+    Args:
+        qubits: List of qubits to be measured.
+        backend (Union[Backend, BaseBackend]): A backend instance, which contains
+            hardware-specific data required for scheduling.
+        target: The :class:`~.Target` representing the target backend, If None, defaults to the
+                  ``target`` of ``backend``.
+        meas_map: List of sets of qubits that must be measured together. If None, defaults to
+                  the ``meas_map`` of ``backend``.
+        qubit_mem_slots: Mapping of measured qubit index to classical bit index.
+        measure_name: Name of the measurement schedule.
+
+    Returns:
+        A measurement schedule corresponding to the inputs provided.
+
+    Raises:
+        PulseError: If both ``target`` and ``backend`` is None.
+    """
+    schedule = Schedule(name=f"Default measurement schedule for qubits {qubits}")
+    try:
+        target = target or backend.target
+        meas_map = meas_map or target.meas_map.get_qubit_groups(qubits)
+    except AttributeError as ex:
+        raise exceptions.PulseError(
+            "inst_map or meas_map, and backend cannot be None simultaneously"
+        ) from ex
+    if isinstance(meas_map, list):
+        meas_map = utils.format_meas_map(meas_map)
+
+    measure_groups = set()
+    for qubit in qubits:
+        measure_groups.add(tuple(meas_map[qubit]))
+    for measure_group_qubits in measure_groups:
+        try:
+            default_sched = target.get_calibration(measure_name, measure_group_qubits)
```

Comment:
This doesn't work because `Target` has calibration for single qubit (because attached `Measure` class is single qubit operation). Here, you need to iterate over all qubits in the group and combine them. 

Also, I have been wondering if grouping is really needed without constraints. For example, if you combine meas calibration for group of [0, 1] and [2, 3], this will eventually generate a left-aligned block of [0, 1, 2, 3]. I think we just need to combine them without considering the grouping, as long as backend meas map doesn't provide constraint. Probably, we need to pad non-measured qubit timeslot in the group with delays (so that they don't cause overlap within the group). Any thoughts @wshanks ?